datadog_checks_downloader/tests/local_http.py

```diff
@@ -0,0 +1,90 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import contextlib
+import logging
+import os
+import shutil
+import tempfile
+import time
+import zipfile
+from functools import partial
+from threading import Thread
+
+import requests
+import six
+
+if six.PY3:
+    from http.server import SimpleHTTPRequestHandler
+    from queue import Queue
+    from socketserver import TCPServer
+
+else:
+    from Queue import Queue
+    from SimpleHTTPServer import SimpleHTTPRequestHandler
+    from SocketServer import TCPServer
+
+_LOGGER = logging.getLogger(__name__)
+_E2E_TESTS_DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")
+_DEFAULT_PORT = 8080
+
+
+class _CustomTCPServer(TCPServer):
+    """A custom TCP server that reuses address."""
+
+    allow_reuse_address = True
+
+
+def _do_local_http_server(queue, directory, port):
+    """Serve requests in a separate thread."""
+    curr_dir = os.getcwd()
+    os.chdir(directory)
+    try:
+        httpd = _CustomTCPServer(("", port), partial(SimpleHTTPRequestHandler))
+        queue.put(httpd)
+        httpd.serve_forever()
+        queue.task_done()
+    finally:
+        os.chdir(curr_dir)
+
+
+@contextlib.contextmanager
+def local_http_server(test_file_case, port=_DEFAULT_PORT):
+    """Start a local HTTP server for E2E tests."""
+    zip_file_path = os.path.join(_E2E_TESTS_DATA_DIR, test_file_case + ".zip")
+    served_dir = tempfile.mkdtemp(prefix=test_file_case)
+
+    try:
+        with zipfile.ZipFile(zip_file_path) as tests_zip_file:
+            tests_zip_file.extractall(path=served_dir)
```

Comment:
In [this pull request I am working on](https://github.com/DataDog/integrations-core-vault-ceremony/pull/64) I would like to be able to invoke the datadog checks downloader pointing it at an arbitrary directory (the one produced by the ceremony I want to test).

This here code will not work for my usecase because I will probably want to manage the served directory myself and simply provide its path through `served_dir`.

Do you think it would be possible to move the first few lines to another function so that `local_http_server` takes `served_dir` as input?

Reply:
That's a good point, I forgot about this use case. Change in 52a28b69fc addresses this, also documented in the README file.